instrumentation/servlet/servlet-3.0/javaagent/src/test/groovy/JettyServlet3Test.groovy

```diff
@@ -155,8 +158,9 @@ class JettyServlet3TestFakeAsync extends JettyServlet3Test {
 
   @Override
   boolean testException() {
-    // https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/807
-    return false
+    // we expect this request to fail with http 500 but is succeeds with http 200
+    // when using -PtestLatestDeps=true
```

Comment:
do u know if this is a limitation of jetty, our instrumentation, or our tests?

Reply:
It is hard to say. My guess is that our test is weird and could very well depend on unspecified behaviour. Test first completes an AsyncContext and then lets exception propagate out of servlet. It might be better to not complete AsyncContext and rely on timeout (if AsyncContext isn't completed container will close it with timeout). In that case the behaviour should be better defined, but timeout has its own issues. There is always the problem that we'd need to make the timeout as short as possible to not waste time but still have it long enough so that test doesn't timeout before the exception is thrown even when running under load.
The tests that are currently failing also exhibit weirdness, apparently on tomcat10 exception test can break the subsequent test.